Content.Shared/RemoteControl/SharedRemoteControlSystem.cs

```diff
@@ -0,0 +1,247 @@
+using Content.Shared.Actions;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Mind;
+using Content.Shared.Mind.Components;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Popups;
+using Content.Shared.RemoteControl.Components;
+using Content.Shared.SSDIndicator;
+using Content.Shared.Verbs;
+
+namespace Content.Shared.RemoteControl;
+
+/// <summary>
+/// System used for managing remote control: Granting temporary control of entities to other entities.
+/// </summary>
+public abstract class SharedRemoteControlSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actions = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RemotelyControllableComponent, MapInitEvent>(OnControllableInit);
+        SubscribeLocalEvent<RemotelyControllableComponent, ComponentShutdown>(OnControllableShutdown);
+        SubscribeLocalEvent<RemotelyControllableComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<RemotelyControllableComponent, AfterInteractUsingEvent>(OnAfterInteractUsing);
+        SubscribeLocalEvent<RemotelyControllableComponent, RemoteControlReturnToBodyEvent>(OnReturnToBody);
+        SubscribeLocalEvent<RemotelyControllableComponent, MobStateChangedEvent>(OnMobStateChanged);
+
+        SubscribeLocalEvent<RemoteControllerComponent, DamageChangedEvent>(OnTookDamage);
+        SubscribeLocalEvent<RemoteControllerComponent, InteractionSuccessEvent>(OnSuccessfulInteract);
+
+        SubscribeLocalEvent<RCRemoteComponent, GetVerbsEvent<ActivationVerb>>(OnRCRemoteVerbs);
+        SubscribeLocalEvent<RCRemoteComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<RCRemoteComponent, ComponentShutdown>(OnRemoteShutdown);
+        SubscribeLocalEvent<RCRemoteComponent, DroppedEvent>(OnRemoteDropped);
+    }
+
+    private void OnExamine(Entity<RemotelyControllableComponent> ent, ref ExaminedEvent args)
+    {
+        if (ent.Comp.ExamineMessage == null
+            || !args.IsInDetailsRange)
+            return;
+
+        args.PushText(Loc.GetString(ent.Comp.ExamineMessage, ("user", ent.Owner)));
```

Comment:
Does this use their identity?

Reply:
I mean... Does it matter?
![image](https://github.com/user-attachments/assets/5ea165aa-9c02-4f72-a616-2286b4080c2e)
![image](https://github.com/user-attachments/assets/2b47337f-6e64-4571-bcf1-42c0a273f18e)
![image](https://github.com/user-attachments/assets/3351cba7-37c2-4ce0-8c9f-ca765a45a62e)

Reply:
Ah, seems fine then.